sys/net/gnrc/link_layer/lwmac/rx_state_machine.c

```diff
@@ -183,9 +183,7 @@ static bool _send_wa(gnrc_netif_t *netif)
     /* Send WA */
     if (_gnrc_lwmac_transmit(netif, pkt) < 0) {
         LOG_ERROR("ERROR: [LWMAC-rx] Send WA failed.");
-        if (pkt != NULL) {
-            gnrc_pktbuf_release(pkt);
-        }
+        gnrc_pktbuf_release(pkt);
```

Comment:
Which coccinelle config causes this output? I am not sure, if `pkt` *could* be `NULL` here.

Reply:
That's also why I'm unsure of this change. Looking at the code, [this line](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/link_layer/lwmac/rx_state_machine.c#L161) already checks if `pkt` is not null and return if it's null. I don't know if code in between could change that non null pkt value.

Reply:
The variable is not changed after L161, the address of ptr doesn't escape. pkt cannot be NU:: in lin 186.

Reply:
Thanks for confirming @kaspar030. So I guess we are good here ?

Reply:
Agreed. Sorry didn't look at the full code this morning.